parser-typechecker/src/Unison/PrettyPrintEnv.hs

```diff
@@ -34,10 +36,43 @@ instance Show PrettyPrintEnv where
   show _ = "PrettyPrintEnv"
 
 fromNames :: Int -> Names -> PrettyPrintEnv
-fromNames length names = PrettyPrintEnv terms' types' where
-  terms' r = safeHead . Set.map HQ'.toHQ $ (Names.termName length r names)
-  types' r = safeHead . Set.map HQ'.toHQ $ (Names.typeName length r names)
-
+fromNames len names = PrettyPrintEnv terms' types' where
+  terms' r = shortestName . Set.map HQ'.toHQ $ (Names.termName len r names)
+  types' r = shortestName . Set.map HQ'.toHQ $ (Names.typeName len r names)
+  shortestName ns = safeHead $ sortOn (length . HQ.toString) (toList ns)
+
+fromSuffixNames :: Int -> Names -> PrettyPrintEnv
+fromSuffixNames len names = fromNames len (Names.suffixify names)
+
+fromNamesDecl :: Int -> Names -> PrettyPrintEnvDecl
+fromNamesDecl len names = 
+  PrettyPrintEnvDecl (fromNames len names) (fromSuffixNames len names)
+
+-- A pair of PrettyPrintEnvs:
+--   - suffixifiedPPE uses the shortest unique suffix
+--   - unsuffixifiedPPE uses the shortest full name
+data PrettyPrintEnvDecl = PrettyPrintEnvDecl {
+  unsuffixifiedPPE :: PrettyPrintEnv,
+  suffixifiedPPE :: PrettyPrintEnv
+  } deriving Show
```

Comment:
@pchiusano and I discussed looking into an as-of-yet-undetermined `PPE` interface would be clearer to use than the mixed current `PPE` / `PrettyPrintEnvDecl`.  A spec for the tool and pretty printer outputs would probably inform this design, or we could work backwards from the use-site calls we want to the model that neatly supports them.